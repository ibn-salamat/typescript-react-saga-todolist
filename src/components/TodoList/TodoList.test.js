/* eslint-disable react/jsx-pascal-case */
import React from "react";
import { shallow } from "enzyme";

import { _TodoList } from ".";
import { CInput } from "..";

const setup = (props) => shallow(<_TodoList {...props} />);
const todos = [{ title: "todo1", id: 1, createdDate: new Date(), done: false }];

describe("TodoList", () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, "useState");
  useStateSpy.mockImplementation((init) => [init, setState]);

  it("should render component", () => {
    const component = setup();
    expect(component).toMatchSnapshot();
  });

  it("should render todolist", () => {
    const component = setup({ todos });
    expect(component).toMatchSnapshot();

    expect(component.find(".todo-container p").text()).toBe(
      `${todos[0].title} | ${todos[0].createdDate.toISOString()}`
    );
  });

  it("should render title", () => {
    const title = "My Title";

    const componentWithoutTitle = setup();
    expect(componentWithoutTitle.find("Title").prop("label")).toBe(undefined);

    const componentWithTitle = setup({ title });
    expect(componentWithTitle.find("Title").prop("label")).toBe(title);
  });
});
