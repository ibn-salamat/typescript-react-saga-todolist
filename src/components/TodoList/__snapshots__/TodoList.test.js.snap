// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should render component 1`] = `
<div
  className="todolist-container"
>
  <Title />
  <CInput
    onChange={[Function]}
    value=""
  />
  <CButton
    disabled={true}
    onClick={[Function]}
  >
    Create
  </CButton>
  <hr />
</div>
`;

exports[`TodoList should render todolist 1`] = `
<div
  className="todolist-container"
>
  <Title />
  <CInput
    onChange={[Function]}
    value=""
  />
  <CButton
    disabled={true}
    onClick={[Function]}
  >
    Create
  </CButton>
  <div
    className="todo-container"
    key="1"
  >
    <p>
      todo1
       | 
      2021-09-28T11:54:37.621Z
    </p>
    isDone: 
  </div>
  <hr />
</div>
`;
